src/dotty/tools/dotc/transform/LambdaLift.scala

```diff
@@ -102,9 +102,10 @@ class LambdaLift extends MiniPhase with IdentityDenotTransformer { thisTransform
 
     /** Mark symbol `sym` as being free in `enclosure`, unless `sym`
      *  is defined in `enclosure` or there is a class between `enclosure`s owner
-     *  and the owner of `sym`.
-     *  Return `true` if there is no class between `enclosure` and
-     *  the owner of sym.
+     *  and the owner of `sym`. Also, update lifted owner of `enclosure` so
```

Comment:
I'd better change `lifted owner of enclosure` to 
`value of liftedOwner(enclosure)`, to make it clear that the map is updated, not the symbol.